src/components/Beacon/Beacon.js

```diff
@@ -0,0 +1,403 @@
+const BEACON_EMBED =
+  '!function(e,t,n){function a(){var e=t.getElementsByTagName("script")[0],n=t.createElement("script");n.type="text/javascript",n.async=!0,n.src="https://beacon-v2.helpscout.net",e.parentNode.insertBefore(n,e)}if(e.Beacon=n=function(t,n,a){e.Beacon.readyQueue.push({method:t,options:n,data:a})},n.readyQueue=[],"complete"===t.readyState)return a();e.attachEvent?e.attachEvent("onload",a):e.addEventListener("load",a,!1)}(window,document,window.Beacon||function(){});'
+const BEACON_INIT =
+  "window.Beacon('init', '163e0284-762b-4e2d-b3b3-70a73a7e6c9f')"
+const CLOSED = 'closed, user can open opt-in dialogue'
+const OPENED = 'opened, user can opt-in or close'
+
+const Beacon = React.memo(({ optedIn = false, onOptIn }) => {
+  return (
+    <div
+      css={`
+        position: absolute;
+        bottom: ${2 * GU}px;
+        right: ${2 * GU}px;
+        z-index: 1001;
+
+        ${breakpoint(
+          'medium',
+          `
+            z-index: 10000;
+            bottom: ${3 * GU}px;
+            right: ${3 * GU}px;
+          `
+        )}
+      `}
+    >
+      {optedIn && (
+        <Helmet>
+          <script type="text/javascript">{BEACON_EMBED}</script>
+          <script type="text/javascript">{BEACON_INIT}</script>
+          <style>
+            {`
+              .BeaconFabButtonFrame,
+              .c-BeaconCloseButton,
+              #beacon-container .Beacon div:first-of-type {
+                display: none !important;
+              }
+              #beacon-container .BeaconContainer {
+                height: calc(100vh - 90px - 48px) !important;
+                max-height: calc(100vh - 90px - 48px) !important;
+                width: calc(100vw - 32px) !important;
+                top: ${6 * GU}px !important;
+                left: ${2 * GU}px !important;
+              }
+              @media (min-width: 768px) {
+                #beacon-container .BeaconContainer {
+                  height: 600px !important;
+                  width: 350px !important;
+                  top: unset !important;
+                  left: unset !important;
+                  bottom: 100px !important;
+                  right: ${3 * GU}px !important;
+                }
+              }
+            `}
+          </style>
+        </Helmet>
+      )}
+      <HelpOptIn onOptIn={onOptIn} optedIn={optedIn} />
+    </div>
+  )
+})
+
+Beacon.propTypes = {
+  optedIn: PropTypes.bool,
+  onOptIn: PropTypes.func.isRequired,
+}
+
+Beacon.defaultProps = {
+  load: false,
+}
+
+const HelpOptIn = React.memo(({ onOptIn, optedIn }) => {
+  const [mode, setMode] = React.useState(CLOSED)
+
+  const handleClose = React.useCallback(() => setMode(CLOSED))
+  const handleToggle = React.useCallback(() => {
+    setMode(mode === CLOSED ? OPENED : CLOSED)
+    if (optedIn) {
+      window.Beacon('toggle')
+    }
+  })
+  const handleOptIn = React.useCallback(() => {
+    onOptIn()
+    setTimeout(() => window.Beacon('open'), 100)
```

Comment:
is the timeout needed for `window.Beacon` to load? Can you add a comment about that? Also, since it's not a sure thing. Perhaps it's sensible to check if Beacon is truthy before calling?

Reply:
Good comment/question: there were different behaviours when implementing this feature, when the user opts in this makes the modal open for the first time. With my current network settings and different browers this is the minimum time I offsetted to make sure the modal opens up on that first-opt-in click.
There is no `onload` trigger to listen to (the 1st scripts loads a 2nd one that then loads a 3rd one),